### PR TITLE
Remove vlan.localAreas

### PR DESCRIPTION
**Pull request by yaocw2020**
Related issue: https://github.com/harvester/harvester/issues/2903

### CODE REVIEW

**Review by w13915984028 — APPROVED** [review]
LGTM, thanks.

File: pkg/controller/agent/vlanconfig/controller.go [hunk 1]
```diff
@@ -166,12 +166,15 @@ func (h Handler) setupVLAN(vc *networkv1.VlanConfig) error {
 		goto updateStatus
 	}
 	// set up VLAN
-	v = vlan.NewVlan(vc.Spec.ClusterNetwork, localAreas)
-	setupErr = v.Setup(uplink)
+	v = vlan.NewVlan(vc.Spec.ClusterNetwork)
+	if setupErr = v.Setup(uplink); setupErr != nil {
+		goto updateStatus
```
Comment (guangbochen) [inline pkg/controller/agent/vlanconfig/controller.go:171]: a `goto` method is preferred when the normal method does not apply to you, in here I would suggest us to refactor to call a normal method in the future version.

File: pkg/controller/agent/vlanconfig/controller.go [hunk 2]
```diff
@@ -166,12 +166,15 @@ func (h Handler) setupVLAN(vc *networkv1.VlanConfig) error {
...
+	}
+	setupErr = h.AddLocalAreas(v, localAreas)
```
Comment (guangbochen) [inline pkg/controller/agent/vlanconfig/controller.go:173]: whey do we ignore the setupErr here?

Comment (yaocw2020) [reply pkg/controller/agent/vlanconfig/controller.go:173]: We don't ignore it. The follow codes is to handle the error when updating the status.

File: pkg/controller/agent/vlanconfig/controller.go (hunk 1 above)
Comment (yaocw2020) [reply pkg/controller/agent/vlanconfig/controller.go:171]: OK

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**